core/src/main/kotlin/org/evomaster/core/problem/rest/classifier/probabilistic/AbstractProbabilistic400EndpointModel.kt

```diff
@@ -54,33 +56,44 @@ abstract class AbstractProbabilistic400EndpointModel(
      * Updating classifier performance based on its prediction
      * Before the warmup is completed, the update is based on a crude guess (like a coin flip).
      */
-    protected fun updatePerformance(input: RestCallAction, outputStatusCode: Int?) {
+    protected fun updateModelMetrics(action: RestCallAction, result: RestCallResult) {
 
-        if (modelAccuracyFullHistory.totalSentRequests < warmup || input.parameters.isEmpty()) {
-            val guess = randomness.nextBoolean()
-            modelAccuracyFullHistory.updatePerformance(guess)
-            modelAccuracy.updatePerformance(guess)
+        val outputStatusCode= result.getStatusCode()
+        if (modelMetricsFullHistory.totalSentRequests < warmup || action.parameters.isEmpty()) {
+            val predictedStatusCode = if(randomness.nextBoolean()) 400 else 200
+            modelMetricsFullHistory.updatePerformance(predictedStatusCode, outputStatusCode?:-1)
+            modelMetrics.updatePerformance(predictedStatusCode, outputStatusCode?:-1)
         } else {
-            val predicted = classify(input).prediction()
-            val predictIsCorrect = (predicted == outputStatusCode)
-            modelAccuracyFullHistory.updatePerformance(predictIsCorrect)
-            modelAccuracy.updatePerformance(predictIsCorrect)
+            val predictedStatusCode = classify(action).prediction()
+            modelMetricsFullHistory.updatePerformance(predictedStatusCode, outputStatusCode?:-1)
+            modelMetrics.updatePerformance(predictedStatusCode, outputStatusCode?:-1)
         }
 
     }
 
-
-    /** Default accuracy estimates */
-    override fun estimateAccuracy(endpoint: Endpoint): Double {
+    /** Default metrics estimates */
+    override fun estimateMetrics(endpoint: Endpoint): ModelEvaluation {
         verifyEndpoint(endpoint)
-        return estimateOverallAccuracy()
+        if (!initialized) {
+            // hasn’t learned anything yet → return defaults
+            return ModelEvaluation.DEFAULT_NO_DATA
+        }
+
+        return ModelEvaluation(
+            modelMetrics.estimateAccuracy(),
+            modelMetrics.estimatePrecision400(),
+            modelMetrics.estimateRecall400(),
+            modelMetrics.estimateMCC400()
+        )
```

Comment:
same as in other case. a model for a specific endpoint will have no difference between `estimateMetrics(endpoint: Endpoint)` and `estimateOverallMetrics()` apart from the endpoint validation in the former. so one function should call the other to avoid code duplication. also, what is the difference between:
```
return ModelEvaluation(
            modelMetrics.estimateAccuracy(),
            modelMetrics.estimatePrecision400(),
            modelMetrics.estimateRecall400(),
            modelMetrics.estimateMCC400()
        )
```
and
```
return modelMetrics.estimateMetrics()
```
?